File: librored/backend/src/main/java/es/codeurjc13/librored/dto/BookDTO.java
package es.codeurjc13.librored.dto;

import es.codeurjc13.librored.model.Book;

/**
 * Book DTO for REST API operations
 * Contains all book information including owner reference
 */
public record BookDTO(
        Long id,
        String title,
        String author,
        Book.Genre genre,
        String description,
        boolean hasCoverImage,
        UserBasicDTO owner) {
}

File: librored/frontend/src/app/dtos/book.dto.ts
export interface BookDTO {
  id?: number;
  title: string;
  description: string;
  author?: string;
  genre?: string;
  hasCoverImage: boolean;
  owner?: {
    id: number;
    username: string;
  };
  shops?: ShopBasicDTO[];
}

export interface ShopBasicDTO {
  id: number;
  name: string;
  address: string;
}